src/main/java/tool/designpatterns/verifiers/multiclassverifiers/proxy/ProxyFeedbackUtils.java

```diff
@@ -0,0 +1,373 @@
+     *
+     * @return a feedback on how they are used.
+     */
+    public static Feedback allClassesAreUsed(
+        List<ProxyPatternGroup> proxyPatternGroups, List<ClassOrInterfaceDeclaration> proxies,
+        List<ClassOrInterfaceDeclaration> interfaces, List<ClassOrInterfaceDeclaration> subjects,
+        List<PartialProxyImplementation> proxyGroups,
+        List<PartialProxyImplementation> subjectGroups) {
+
+        // Remove the classes used in valid instances from the maps.
+        List<ClassOrInterfaceDeclaration> unusedProxies = getUnused(proxies, proxyPatternGroups,
+            Pattern.PROXY_PROXY);
+        List<ClassOrInterfaceDeclaration> unusedInterfaces = getUnused(interfaces,
+            proxyPatternGroups, Pattern.PROXY_INTERFACE);
+        List<ClassOrInterfaceDeclaration> unusedSubjects = getUnused(subjects, proxyPatternGroups,
+            Pattern.PROXY_SUBJECT);
+
+        List<PartialProxyImplementation> partialProxyGroups = getCorrectPartial(proxyGroups,
+            proxyPatternGroups, Pattern.PROXY_PROXY);
+        List<PartialProxyImplementation> partialSubjectGroups = getCorrectPartial(subjectGroups,
+            proxyPatternGroups, Pattern.PROXY_SUBJECT);
+
+        Tuple2<List<ProxyPatternGroup>, List<ProxyPatternGroup>> validInvalidTuple
+            = splitToValidInvalid(proxyPatternGroups);
+        List<ProxyPatternGroup> valid = validInvalidTuple.getFirst();
+        List<ProxyPatternGroup> invalid = validInvalidTuple.getSecond();
+
+        List<Feedback> feedbacks = getInvalidFeedback(invalid, valid);
+
+        // Now add feedback for those classes that are still not used.
+        feedbacks.add(
+            getUnusedFeedbacks(unusedProxies, unusedInterfaces, unusedSubjects, partialProxyGroups,
+                partialSubjectGroups));
+
+        return Feedback.getPatternInstanceFeedback(feedbacks);
+    }
+
+    /**
+     * Finds the unused classes for a specific pattern.
+     *
+     * @param all     all classes marked with the pattern.
+     * @param valid   all the valid ProxyPatternGroups found.
+     * @param pattern the pattern.
+     *
+     * @return a new list of unused classes of the given pattern.
+     */
+    private static List<ClassOrInterfaceDeclaration> getUnused(
+        List<ClassOrInterfaceDeclaration> all, List<ProxyPatternGroup> valid, Pattern pattern) {
+
+        List<ClassOrInterfaceDeclaration> unused = new ArrayList<>();
+        for (ClassOrInterfaceDeclaration classOrI : all) {
+            boolean used = false;
+            String classQualName = classOrI.resolve().getQualifiedName();
+            // We cannot use .contains due to the fact that it can add the wrong class if
+            // identical classes are in different packages for example.
+            for (ProxyPatternGroup group : valid) {
+                String currQualName = group.getFromPattern(pattern).resolve().getQualifiedName();
+                if (classQualName.equals(currQualName)) {
```

Comment:
```suggestion
            // We cannot use .contains due to the fact that it can add the wrong class if
            // identical classes are in different packages for example.
            for (ProxyPatternGroup group : valid) {
                if (isSameClassOrInterfaceDeclaration(classOrI, group.getFromPattern(pattern))) {
```

Reply:
Solved in ef5b269